api/redhatopenshift/Hypershift/hcpCluster-models.tsp

```diff
@@ -0,0 +1,547 @@
+   * - https://docs.openshift.com/container-platform/4.12/networking/openshift_sdn/about-openshift-sdn.html
+  */
+  OpenShiftSDN:  "OpenShiftSDN",
+  /** THE OVN network plugin for the OpenShift cluster
+   * - https://docs.openshift.com/container-platform/4.12/networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.html
+   */
+  OVNKubernetes: "OVNKubernetes",
+  /** Other network plugins */
+  Other:         "Other",
+}
+
+/** Configuration of the cluster web console */
+model ConsoleProfile {
+  /** The cluster web console URL endpoint */
+  @visibility("read")
+  url: string;
+}
+
+/** Information about the API of a cluster. */
+model ApiProfile {
+  /** URL endpoint for the API server */
+  @visibility("read")
+  url: string;
+
+  /** ip address of the API server */
+  @visibility("read")
+  ip: string;
+
+  /** should the API server be accessible from the internet */
+  @visibility("create")
+  visibility: Visibility;
+}
+
+/** The visibility of the API server */
+enum Visibility {
+  /** The API server is visible from the internet. */
+  public: "public",
+
+  /** The API server is not visible from the internet. */
+  private: "private",
+}
+
+/** OpenShift cluster proxy configuration
+ * - https://docs.openshift.com/container-platform/4.12/networking/enable-cluster-wide-proxy.html
+*/
+model ProxyProfile {
+  /** http proxy config */
+  httpProxy?: string;
+
+  /** https proxy config */
+  httpsProxy?: string;
+
+  /** no proxy config */
+  noProxy?: string;
+
+  /** The trusted CA for the proxy */
+  trustedCa?: string;
+}
+
+/** Azure specific configuration */
+model PlatformProfile {
+  /** Resource group to put cluster resources */
+  managedResourceGroup: string;
+
+  /** ResourceId for the subnet used by the control plane */
+  subnetName: string;
+
+  /** The core outgoing configuration */
+  outboundType?: OutboundType = OutboundType.loadBalancer;
+
+  /** Specifies whether subnets are pre-attached with an NSG */
+  preconfiguredNsgs: boolean;
+
+  /** The id of the disk encryption set to be used for etcd.
+   * Configure this when `etcdEncryption` is set to true
+   * Is used the https://learn.microsoft.com/en-us/azure/storage/common/customer-managed-keys-overview
+  */
+  etcdEncryptionSetId?: string;
+}
+
+/** The outbound routing strategy used to provide your cluster egress to the internet. */
+enum OutboundType {
+  /** The loadbalancer configuration */
+  loadBalancer: "loadBalancer",
+}
+
+/** Etcd encryption profile sets the configuration needed for
+ * customer brought keys to encrypt the etcd.
+ * This is an optional field, if omitted the default encryption key is used.
+ * TODO: decide how this is going to be implemented
+*/
+model EtcdEncryptionProfile {
+  /** The id of the disk encryption set to be used for etcd
+   * Is relies on https://learn.microsoft.com/en-us/azure/storage/common/customer-managed-keys-overview
+  */
+  discEncryptionSetId: string;
+}
```

Comment:
This is no longer used, since the `etcdEncryptionSetId` is now part of the `PlatformProfile` model.